tests/test_unordered.py

```diff
@@ -7,20 +7,85 @@
 
 
 @pytest.mark.parametrize(
-    "left,right",
+    ["expected", "actual"],
     [
         (unordered(1, 2, 3), [3, 2, 1]),
-        ([3, 2, 1], unordered(1, 2, 3)),
+        (unordered(1, 2, 3), (3, 2, 1)),
+        (unordered(1, 2, 3), {3, 2, 1}),
+        (unordered([1, 2, 3]), [3, 2, 1]),
+        (unordered((1, 2, 3)), (3, 2, 1)),
+        (unordered({1, 2, 3}), {3, 2, 1}),
+        (unordered({1: 2, 3: 4}), {3: 4, 1: 2}),
         (unordered(1, 2, {"a": unordered(4, 5, 6)}), [{"a": [6, 5, 4]}, 2, 1]),
-        ([3, 2, {1: ['b', 'a']}], unordered({1: unordered('a', 'b')}, 2, 3)),
+        (unordered([{1: unordered(['a', 'b'])}, 2, 3]), [3, 2, {1: ['b', 'a']}]),
         (unordered(x for x in range(3)), [2, 1, 0]),
-        ((x for x in range(3)), unordered(2, 1, 0)),
+        (unordered(x for x in range(3)), (2, 1, 0)),
+        (unordered(x for x in range(3)), {2, 1, 0}),
+        (unordered(x for x in range(3)), range(3)),
+        (unordered("abc"), "bac"),
+        (unordered("a", "b", "c"), ["b", "a", "c"]),
+        (unordered("a", "b", "c"), "bac"),
     ],
 )
-def test_unordered(left, right):
+def test_unordered(expected, actual):
+    assert expected == actual
+    assert actual == expected
+    with pytest.raises(AssertionError):
+        assert expected != actual
```

Comment:
Why not use `assert not (expected != actual)` here?

Reply:
I wanted to check the assertion message through `raises( ... match=...)` but that turned out not very interesting. I can indeed do it with `assert not`